### PR TITLE
lief: support conan v2 and update dependencies

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **lief/0.10.1**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
@uilianries 
Thanks a lot!

File: recipes/lief/all/conanfile.py [hunk 1]
```diff
@@ -42,80 +47,101 @@ class LiefConan(ConanFile):
...
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+
+        if self.info.settings.compiler == "Visual Studio" and Version(self.info.settings.compiler.version) <= "14" and self.options.shared:
```
Comment (prince-chrismc) [inline recipes/lief/all/conanfile.py:72]: missing msvc?

File: recipes/lief/all/conanfile.py [hunk 2]
```diff
@@ -42,80 +47,101 @@ class LiefConan(ConanFile):
...
-        self.cpp_info.names["pkg_config"] = "lief"
-        self.cpp_info.names["cmake_find_package"] = "LIEF"
-        self.cpp_info.names["cmake_find_package_multi"] = "LIEF"
-        self.cpp_info.libs = tools.collect_libs(self)
-        self.cpp_info.defines = ["_GLIBCXX_USE_CXX11_ABI=1"]
+        self.cpp_info.libs = ["LIEF"]
+
+        self.cpp_info.set_property("cmake_file_name", "LIEF")
+        self.cpp_info.set_property("cmake_target_name", "LIEF::LIEF")
+        self.cpp_info.set_property("pkg_config_name", "LIEF")
```
Comment (prince-chrismc) [inline recipes/lief/all/conanfile.py:131]: used to be lowercase?

**Review by prince-chrismc — COMMENTED** [review]
Two tiny questions

File: recipes/lief/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/lief/all/conanfile.py:131]: official package name seems to be LIEF.
https://lief-project.github.io/doc/latest/installation.html#find-package

I added these lines for conan v2.

File: recipes/lief/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/lief/all/conanfile.py:72]: I forgot to support msvc.
Thanks.

Comment (toge) [reply recipes/lief/all/conanfile.py]: fixed

**Comment by toge** [comment]
@uilianries 
I met CI errors in my several PRs as following:

```
  Error: Unable to locate the previous sha: 5a274d88a9fa3e53f925b83bbf9917a27e5e439a
  Error: You seem to be missing 'fetch-depth: 0' or 'fetch-depth: 2'. See https://github.com/tj-actions/changed-files#usage
  Error: Process completed with exit code 1.
```

I don't know what is wrong.
Would you have any information to fix them?

**Comment by uilianries** [comment]
@toge My guess is because we updated GH packages and now it requires some attributes which were not present on all files. I created the PR with a possible fix: https://github.com/conan-io/conan-center-index/pull/13580

File: recipes/lief/all/patches/001_link_to_conan.patch [hunk 3]
```diff
@@ -83,38 +85,42 @@
...
    source_group("easylogging"     FILES ${ELG_SOURCE_DIR}/easylogging++.cc)
  endif()
-@@ -316,7 +303,7 @@
+@@ -316,7 +303,8 @@ target_include_directories(LIB_LIEF
    PRIVATE "${LIEF_PRIVATE_INCLUDE_DIR}")
-
+ 
  if (LIEF_ENABLE_JSON)
 -  add_dependencies(LIB_LIEF lief_libjson)
-+  target_link_libraries(LIB_LIEF CONAN_PKG::nlohmann_json)
```
Comment (uilianries) [inline recipes/lief/all/patches/001_link_to_conan.patch:91]: Is it really required? CONAN_PKG will be deprecated soon.

Comment (toge) [reply recipes/lief/all/patches/001_link_to_conan.patch:91]: I agree with you.
This line is deleted by this PR.

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**